sdk/ai/Azure.AI.Projects/src/Custom/AIProjectClient.cs

```diff
@@ -39,62 +44,58 @@ public AIProjectClient(string connectionString, TokenCredential credential, AIPr
-            // a separate "Serverless" connection. This is now deprecated.
-            bool useServerlessConnection = Environment.GetEnvironmentVariable("USE_SERVERLESS_CONNECTION") == "true";
-            ConnectionType connectionType = useServerlessConnection ? ConnectionType.Serverless : ConnectionType.AzureAIServices;
+        /// <summary>
+        /// Retrieves the connection options for a specified client type and instance ID.
+        /// </summary>
+        public override ClientConnection GetConnection(string connectionId) => _cacheManager.GetConnection(connectionId);
 
-            ConnectionResponse connection = connectionsClient.GetDefaultConnection(connectionType, true);
+        /// <summary>
+        /// Retrieves all connection options.
+        /// </summary>
+        public override IEnumerable<ClientConnection> GetAllConnections() => _cacheManager.GetAllConnections();
 
-            if (connection.Properties is ConnectionPropertiesApiKeyAuth apiKeyAuthProperties)
+        private static TokenCredential BuildCredential(TokenCredential credential)
+        {
+            if (credential != null)
             {
-                if (string.IsNullOrWhiteSpace(apiKeyAuthProperties.Target))
-                {
-                    throw new ArgumentException("The API key authentication target URI is missing or invalid.");
-                }
+                return credential;
+            }
 
-                if (!Uri.TryCreate(apiKeyAuthProperties.Target, UriKind.Absolute, out var endpoint))
-                {
-                    throw new UriFormatException("Invalid URI format in API key authentication target.");
-                }
+            string clientId = Environment.GetEnvironmentVariable("CLOUDMACHINE_MANAGED_IDENTITY_CLIENT_ID");
 
-                var credential = new AzureKeyCredential(apiKeyAuthProperties.Credentials.Key);
-                if (!useServerlessConnection)
-                {
-                    // Be sure to use the Azure resource name here, not the connection name. Connection name is something that
-                    // admins can pick when they manually create a new connection (or use bicep). Get the Azure resource name
-                    // from the end of the connection id.
-                    var azureResourceName = connection.Id.Split('/').Last();
-                    endpoint = new Uri($"https://{azureResourceName}.services.ai.azure.com/models");
-                }
-                return clientFactory(endpoint, credential);
-            }
-            else
-            {
-                throw new ArgumentException("Cannot connect with Inference! Ensure valid ConnectionPropertiesApiKeyAuth.");
-            }
+            return !string.IsNullOrEmpty(clientId)
+                ? new ManagedIdentityCredential(clientId)
+                : new ChainedTokenCredential(new AzureCliCredential(), new AzureDeveloperCliCredential());
```

Comment:
Why isn't this `new DefaultAzureCredential()`?

Reply:
[This](https://learn.microsoft.com/en-us/dotnet/azure/sdk/authentication/credential-chains?tabs=dac#usage-guidance-for-defaultazurecredential) sums it up.

Reply:
Is this what all libraries should be using by default? We are using the `DefaultAzureCredential` in .NET Aspire (basically everywhere) by default if the user doesn't specify it.

Does the above code work when I am just logged into Visual Studio and haven't auth'd with the CLI?

Reply:
> Does the above code work when I am just logged into Visual Studio and haven't auth'd with the CLI?

It should.

Reply:
> Is this what all libraries should be using by default?

Ideally DAC should only be used in dev/test scenarios. It's better to be deterministic when the code is deployed to prod.